src/core/cgroup.c

```diff
@@ -3571,12 +3571,17 @@ void unit_prune_cgroup(Unit *u) {
         if (!crt || !crt->cgroup_path)
                 return;
 
-        /* Cache the last CPU and memory usage values before we destroy the cgroup */
+        /* Cache the last resource usage values before we destroy the cgroup */
         (void) unit_get_cpu_usage(u, /* ret = */ NULL);
 
         for (CGroupMemoryAccountingMetric metric = 0; metric <= _CGROUP_MEMORY_ACCOUNTING_METRIC_CACHED_LAST; metric++)
                 (void) unit_get_memory_accounting(u, metric, /* ret = */ NULL);
 
+        /* All IO metrics are read at once from the underlying cgroup, so issue just a single call */
+        (void) unit_get_io_accounting(u, /* metric = */ 0, /* ret = */ NULL);
```

Comment:
hmm i find it weird to pass 0 here, because that is just a weird way to write `CGROUP_IO_READ_BYTES`, and that's really misleading here.

I think it would be nice to explicitly specify _CGROUP_IO_ACCOUNTING_METRIC_INVALID here, and make sure that unit_get_io_accounting() is fine with that, as long as `ret` is specified as NULL, and in that case just reads stuff and doesn't return anything.

Reply:
Fair enough, done.